docs/pages/api-docs/input.json

```diff
@@ -35,26 +35,7 @@
     "value": { "type": { "name": "any" } }
   },
   "name": "Input",
-  "styles": {
-    "classes": [
-      "root",
-      "formControl",
-      "focused",
-      "disabled",
-      "colorSecondary",
-      "underline",
-      "error",
-      "sizeSmall",
-      "multiline",
-      "fullWidth",
-      "input",
-      "inputSizeSmall",
-      "inputMultiline",
-      "inputTypeSearch"
-    ],
-    "globalClasses": { "focused": "Mui-focused", "disabled": "Mui-disabled", "error": "Mui-error" },
-    "name": "MuiInput"
-  },
+  "styles": { "classes": [], "globalClasses": {}, "name": null },
```

Comment:
Is this okay? 👀

Reply:
The purpose of the PR is to forward the `classes` props to the `InputBase` component so that those can be customized from the other input components `FilledInput`, `Input` and `OutlinedInput`. This is why the types for the `classses` are extended (and the tests are no longer throwing as this is now supported). The only problem I see is the `buildApi` not recognizing this syntax.

Reply:
> The only problem I see is the buildApi not recognizing this syntax.

Thanks for being good faith and ignoring the changed test.

Reply:
> This is why the types for the classses are extended (and the tests are no longer throwing as this is now supported).

Wait, what?

That's definitely not what we want. You shouldn't use props for a variant that you aren't using.

Reply:
> > The only problem I see is the buildApi not recognizing this syntax.
>
> Thanks for being good faith and ignoring the changed test.Thanks for being good faith and ignoring the changed test.

You cannot just copy the last sentence and get a conclusion. I have stated before that:

> The purpose of the PR is to forward the classes props to the InputBase component so that those can be customized from the other input components FilledInput, Input and OutlinedInput. This is why the types for the classses are extended (**and the tests are no longer throwing as this is now supported**)


---

> Then explain that in the original description not after I point it out. 

The PR was initially suppose the change the classes passed, the typings changes were required afterwards, that's why it's not included in the original description.

---


> > why the types for the classses are extended (and the tests are no longer throwing as this is now supported).
>
> Wait, what?
> 
> That's definitely not what we want. You shouldn't use props for a variant that you aren't using.

I did not just blindly suggest this. If you see what `filledProps` is, **it has nothing specific to the `FilledInput`**, the `inputAdornedStart` is supported in InputBase, hence in all Input components. It may have been the case that in the past this was not true, but it is now, so it may be that the test is not up to date.